var/spack/repos/builtin/packages/py-igraph/package.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyIgraph(PythonPackage):
+    """Python interface to the igraph high performance graph library,
+    primarily aimed at complex network research and analysis."""
+
+    homepage = "https://github.com/igraph/python-igraph"
+    pypi = "igraph/igraph-0.10.6.tar.gz"
+
+    version("0.10.6", sha256="76f7aad294514412f835366a7d9a9c1e7a34c3e6ef0a6c3a1a835234323228e8")
+
+    variant("matplotlib", default=False, description="Enable plotting with Matplotlib")
+
+    depends_on("arpack-ng")
+    depends_on("cmake")
+    depends_on("gmp")
+    depends_on("glpk+gmp@4.57:")
+    depends_on("libxml2")
+    depends_on("py-setuptools", type=("build", "run"))
+    depends_on("py-texttable@1.6.2:", type=("build", "run"))
+    depends_on("py-matplotlib@3.5.0:3.6.0", type=("run"), when="+matplotlib")
```

Comment:
```suggestion
    depends_on("py-matplotlib@3.5", type="run", when="+matplotlib")
```
Spack version ranges are inclusive, 3.6.0 shouldn't be included